src/components/switch/style.ts

```diff
@@ -22,38 +23,42 @@ export const StyledSwitch = styled.label`
     top: 0;
     -webkit-transition: 0.4s;
     transition: 0.4s;
+    padding: 2px;
+    // border: 2px solid transparent;
   }
 
-  & .slider::before {
+  & .round {
     background-color: white;
-    bottom: 1px;
+    border-radius: 50%;
     content: '';
-    height: ${(props: any): string => props.height || '16px'};
-    left: 1px;
+    height: calc(${(props: any): string => props.height} - 4px);
+    left: 2px;
     position: absolute;
     -webkit-transition: 0.4s;
     transition: 0.4s;
-    width: ${(props: any): string => props.height || '16px'};
+    width: calc(${(props: any): string => props.height} - 4px);
+    display: flex;
+    justify-content: center;
+    align-items: center;
   }
 
   & input:checked + .slider {
     background-color: ${(props: any): string => props.theme.colors.success};
+    color: ${(props: any): string => props.theme.colors.success};
+    -webkit-transition: 0.4s;
+    transition: color ease-out 0.1s;
   }
 
   & input:focus + .slider {
-    // box-shadow: 0 0 1px $secondary;
-  }
-
-  & input:checked + .slider::before {
-    transform: translateX(16px);
+    box-shadow: 0px 0px 0px 2px ${(props: any): string => props.theme.colors.outline};
   }
 
-  /* Rounded sliders */
-  & .slider.round {
-    border-radius: ${(props: any): string => props.height || '16px'};
+  & input:checked + .slider .round {
+    left: calc(100% + 2px - ${(props: any): string => props.height || '20px'});
```

Comment:
You have a default height in the index file, why do you use "OR" operator in style as well?